lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/client/YarnJobSubmissionHandler.java

```diff
@@ -69,7 +72,8 @@
       final ClasspathProvider classpath,
       final ConfigurationSerializer configurationSerializer,
       final JobUploader uploader,
-      @Parameter(JVMHeapSlack.class) final double jvmSlack) throws IOException {
+      @Parameter(JVMHeapSlack.class) final double jvmSlack,
+      @Parameter(JobQueue.class) String defaultQueueName) throws IOException {
```

Comment:
It is given by the user, but has a default value, 'default'. That is the *only* queue the current code submits to. I don't know whether YARN has an API to figure out the default queue. If we want to use it, I think it would be a good idea to do that in a separate JIRA, as it would actually change current behavior without application changes.